src/fault_injector/fault_types/schedule_blocked_fault.py

```diff
@@ -1,23 +1,48 @@
+from src.fault_injector.fault_configurations.schedule_blocked_fault_configuration import (
+    ScheduleBlockedFaultConfiguration,
+)
 from src.fault_injector.fault_types.fault import Fault
+from src.logger.logger import Logger
+from src.spawner.spawner import Spawner
 
 
 class ScheduleBlockedFault(Fault):
     """A fault that blocks a platform"""
 
+    configuration: ScheduleBlockedFaultConfiguration
+    spawner: Spawner
+
+    def __init__(
+        self,
+        configuration,
+        logger: Logger,
+        spawner: Spawner,
+    ):
+        super().__init__(configuration, logger)
+        self.spawner = spawner
+
     def inject_fault(self, tick: int):
         """inject ScheduleBlockedFault into the given component
 
         :param tick: the simulation tick in which inject_fault was called
         :type tick: Integer
         """
+        self.spawner.block_schedule(self.configuration.affected_element_id)
+        self.logger.inject_schedule_blocked_fault(
+            tick=tick,
+            schedule_blocked_fault_configuration=self.configuration.id,
+            affected_element=self.configuration.affected_element_id,
+        )
         # - get schedule by id
```

Comment:
Do you still need the comments? 🤔

Reply:
Good point, that should be documented by the code now. 📜

Reply:
Thanks! 💚